compose/Dockerfile.compiled

```
@@ -0,0 +1,144 @@
+
+# Make uv executable
+RUN chmod 0755 /usr/local/src/uv/uv-x86_64-unknown-linux-gnu/uv
+
+# Create uv symlink
+RUN ln -s /usr/local/src/uv/uv-x86_64-unknown-linux-gnu/uv /usr/local/bin/uv
+
+# Clone testnet-generation-tool.git repository
+WORKDIR /usr/local/src
+RUN git clone --branch main https://github.com/cardano-foundation/testnet-generation-tool.git
+
+# Download testnet-generation-tool dependencies
+WORKDIR /usr/local/src/testnet-generation-tool
+RUN uv sync
+
+# Copy testnet.yaml specification
+COPY --chown=root:root .testnet.yaml /usr/local/src/testnet-generation-tool/testnet.yaml
+
+# Build testnet configuration files
+RUN uv run python3 genesis-cli.py testnet.yaml -o /tmp/testnet -c generate
+
+# Remove dynamic topology.json
+RUN find /tmp/testnet -type f -name 'topology.json' -exec rm -f '{}' ';'
+
+#---------------------------------------------------------------------
+
+FROM docker.io/debian:stable-slim AS main
+
+# Set environment variables
+ENV CARDANO_NODE_SOCKET_PATH="/opt/cardano-node/data/db/node.socket"
+
+# Set time zone
+ENV TZ="UTC"
+RUN ln -snf /usr/share/zoneinfo/${TZ} /etc/localtime && \
+    echo ${TZ} > /etc/timezone
+
+# Install packages required in main stage
+RUN apt update && \
+    apt install -y --no-install-recommends \
+        dnsutils \
+        iproute2 \
+        iputils-ping \
+        jq \
+        less \
+        lsof \
+        netbase \
+        netcat-openbsd \
+        openssl \
+        procps \
+        tcpdump \
+        telnet \
+        vim
+
+# Copy binaries from build stage
+COPY --from=build --chown=root:root /usr/local/bin/cardano-cli /usr/local/bin/cardano-cli
+COPY --from=build --chown=root:root /usr/local/bin/cardano-node /usr/local/bin/cardano-node
+
+# Create cardano group and user
+RUN groupadd --gid 10000 cardano && \
+    useradd --comment 'cardano' --create-home --gid 10000 --password '!' --shell '/bin/bash' --uid 10000 cardano
+
+# Create cardano-node directories
+RUN install --directory --owner=root --group=root --mode=0755 /opt/cardano-node && \
+    install --directory --owner=cardano --group=cardano --mode=0750 /opt/cardano-node/config && \
+    install --directory --owner=cardano --group=cardano --mode=0750 /opt/cardano-node/data && \
+    install --directory --owner=cardano --group=cardano --mode=0750 /opt/cardano-node/data/db && \
+    install --directory --owner=cardano --group=cardano --mode=0750 /opt/cardano-node/log && \
+    install --directory --owner=cardano --group=cardano --mode=0750 /opt/cardano-node/pools
+
+# Copy pool config and key material
+COPY --from=build --chown=cardano:cardano /tmp/testnet/pools /opt/cardano-node/pools
+COPY --from=build --chown=cardano:cardano /tmp/testnet/utxos /opt/cardano-node/utxos
+
+# Copy canary_tx.sh
+COPY --chown=root:root canary_tx.sh /
+RUN chmod 0755 /canary_tx.sh
+
+# Copy cmd.sh
+COPY --chown=root:root cmd.sh /
+RUN chmod 0755 /cmd.sh
+
+USER cardano
+#STOPSIGNAL SIGINT
+
+CMD ["/bin/bash", "/cmd.sh"]
```

The script's she-bang uses `env` as it's executor so I think there's no need to explicitly wrap the execution in `/bin/bash`

```suggestion
CMD ["/cmd.sh"]
```